test/watcher.test.js

```diff
@@ -0,0 +1,140 @@
+const path = require('path');
+const ncc = global.coverage ? require("../src/index") : require("../");
+const fs = require('fs');
+
+// Based on the NodeWatchFileSystem class at:
+// - https://github.com/webpack/webpack/blob/master/lib/node/NodeWatchFileSystem.js
+// which in turn exposes:
+// - https://www.npmjs.com/package/watchpack
+
+class CustomWatchFileSystem {
+  constructor(watchStart, watchEnd) {
+    this.closed = false;
+    // paused allows the watchers to stay open for the next build
+    this.paused = false;
+    this.changeCallback = undefined;
+    this.watchStart = watchStart;
+    this.watchEnd = watchEnd;
+
+    // Webpack requires us to track this stuff
+    this.files = undefined;
+    this.dirs = undefined;
+    this.missing = undefined;
+    this.timestamps = new Map();
+
+    // this will be populated for us by ncc
+    this.inputFileSystem = undefined;
+  }
+  
+  triggerChanges (changed, removed) {
```

Comment:
This function is a convenience hook in this test implementation, allowing us to "mock" a watch change event, where `changed` is an array of file paths changed, and `removed` is an array of file paths deleted. Webpack usually aggregates these over a time interval.